File: src/main/java/com/allscontracting/service/CompanyService.java
package com.allscontracting.service;

import java.util.List;
import java.util.stream.Collectors;

import org.springframework.data.domain.Sort;
import org.springframework.stereotype.Service;

import com.allscontracting.dto.CompanyDTO;
import com.allscontracting.exception.LeadsException;
import com.allscontracting.model.Company;
import com.allscontracting.repo.CompanyRepository;

import lombok.RequiredArgsConstructor;

@Service
@RequiredArgsConstructor
public class CompanyService {

	private final CompanyRepository companyRepo;

	public List<CompanyDTO> getCompanies() {
		return this.companyRepo.findAll(Sort.by("name")).stream().map(c -> CompanyDTO.of(c)).collect(Collectors.toList());
	}

	public CompanyDTO update(CompanyDTO companyDTO) throws LeadsException {
		Company company = this.companyRepo.findById(companyDTO.getId()).orElseThrow(() -> new LeadsException("Person not found"));
		company.setAddress(companyDTO.getAddress());
		company.setEmail(companyDTO.getEmail());
		company.setName(companyDTO.getName());
		company.setWebsite(companyDTO.getWebsite());
		return CompanyDTO.of(this.companyRepo.save(company));
	}

	public CompanyDTO save(CompanyDTO companyDTO) {
		Company company = CompanyDTO.toCompany(companyDTO);
		return CompanyDTO.of(this.companyRepo.save(company));
	}

	public List<CompanyDTO> search(String text) {
		return CompanyDTO.of(this.companyRepo.search(text));
	}

}

File: src/main/resources/static/app/service/company-service.js
var local_server_url = "";

 var companyService = function($http){
	 return {

		 search: function (text) {
			 return $http.get(local_server_url + "/companies/search/"+text);
		 },

		 findCompanies: function (scope) {
			 $http.get(local_server_url + "/companies").then(function (response) {
				 scope.companies = response.data.companies
			 }, function (response) {
				 console.log(response)
			 });
		 },
 		
 		update: function (scope, company) {
 			$http.put(local_server_url + "/companies", company).then(function (response) {
 				if(response.data.errorMessages){
 					scope.errorMessages = response.data.errorMessages
 				}else{
 					company = response.data.company
 					scope.successMessages = response.data.successMessages
 				}
 			}, function (response) {
 				console.log(response)
 			});
 		},
 		
 		save: function (scope, company) {
 			$http.post(local_server_url + "/companies", company).then(function (response) {
 				if(response.data.errorMessages){
 					scope.errorMessages = response.data.errorMessages
 				}else{
 					scope.companies.push(response.data.company)
 					scope.successMessages = response.data.successMessages
 				}
 			}, function (response) {
 				console.log(response)
 			});
 		},

 		
 	} 
 }
 
 export default companyService